lib/lcevc/lcevc_dil.js

```diff
@@ -0,0 +1,281 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.lcevc.Dil');
+goog.require('shaka.log');
+goog.require('shaka.util.Dom');
+
+/**
+ * @export
+ * @summary
+ * lcevcDil provides all the operations related to the enhancement and rendering
+ * of LCEVC enabled streams and on to a canvas.
+ */
+shaka.lcevc.Dil = class {
+  /**
+   * @param {HTMLMediaElement} media The video element that will be attached to
+   * LCEVC Dil for input.
+   * @param {HTMLElement} canvas The canvas element that will be attached to
+   * LCEVC Dil to render the enhanced frames.
+   * @param {Object} dilConfig The LCEVC DIL config object
+   * to initialize the LCEVC DIL.
+   */
+  constructor(media, canvas, dilConfig) {
+    /** @private {?LcevcDil.LcevcDIL} */
+    this.dil_ = null;
+
+    /** @private {?Element} */
+    this.parent_ = null;
+
+    /** @private {?HTMLElement} */
+    this.lcevcContainer_ = null;
+
+    /** @private {?HTMLElement} */
+    this.videoControls_ = null;
+
+    /** @private {ResizeObserver} */
+    this.resizeObserver_ = null;
+
+    this.level_ = -1;
+    this.containerFormat_ = 0;
+    this.media_ = /** @type {HTMLVideoElement} */ (media);
+    this.canvas_ = /** @type {HTMLCanvasElement} */ (canvas);
+    this.dilConfig_ = dilConfig;
+
+    // This line is optional forcing the config to hide the LCEVC watermark
+    // that is displayed on the left hand side top corner.
+    if (!this.dilConfig_['logo']) {
+      this.dilConfig_['logo'] = false;
+    }
+  }
+
+  /**
+   * Append data to the LCEVC Dil.
+   * @param {BufferSource} data
+   */
+  appendBuffer(data) {
+    if (this.dil_) {
+      this.dil_.appendBuffer(data, 'video', this.level_);
+    }
+  }
+
+  /**
+  * Hide the canvas and show original video element
+  */
+  hideCanvas() {
+    if (this.dil_) {
+      this.media_.style.display = 'block';
+      this.canvas_.style.display = 'none';
+    }
+  }
+
+  /**
+    * Show the canvas and hide original video element
+    */
+  showCanvas() {
+    if (this.dil_) {
+      this.media_.style.display = 'none';
+      this.canvas_.style.display = 'block';
+    }
+  }
+
+  /**
+   * Create LCEVC Dil.
+   * @private
+   */
+  createDil_() {
+    if (this.isSupported() && !this.dil_) {
+      if (!this.canvas_) {
+        this.createCanvas_();
+      }
+      if (LcevcDil.SupportObject.webGLSupport(this.canvas_)) {
+        this.dil_ = new LcevcDil.LcevcDil(
+            this.media_,
+            this.canvas_,
+            this.dilConfig_);
+
+        this.setControlsStyle_();
+      }
+    }
+  }
+
+  /**
+   * Reset LCEVC Dil.
+   */
+  async reset() {
```

Comment:
Please document return type (`!Promise`)

Reply:
This was not needed anymore, hence removed.